app/src/main/java/com/example/tasky/MainActivity.kt

```diff
@@ -0,0 +1,80 @@
+package com.example.tasky
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import kotlinx.coroutines.delay
+
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            setContent {
+                MyApp()
+            }
+        }
+    }
+}
+
+@Composable
+fun SplashScreen(onSplashComplete: () -> Unit) {
+        LaunchedEffect(key1 = true) {
+        delay(2000) // Delay for 2 seconds
+        onSplashComplete() // Notify that splash is complete
+    }
+    Box(
+        modifier = Modifier
+            .fillMaxSize() // Fill the entire screen
+            .background(colorResource(id = R.color.primary_color)), // Set the background color
+        contentAlignment = Alignment.Center
+    ) {
+        Image(
+            painter = painterResource(id = R.drawable.ic_tasky_logo),
+            contentDescription = null,
+            modifier = Modifier.size(200.dp)
+        )
+    }
+}
+
+@Composable
+fun MainContent() {
+    Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+        Text(text = "Main Content", fontSize = 24.sp)
+    }
+}
+
+@Composable
+fun MyApp() {
+    var showSplashScreen by remember { mutableStateOf(true) }
```

Comment:
This state would also reset after a config change like a screen rotation, so the splash screen would show up again. You could use `rememberSaveable` to avoid that, but I'd rather recommend to keep all UI state in ViewModels to keep responsibilities separate